providers/os/connection/docker/docker_file_connection.go

```diff
@@ -63,6 +63,10 @@ func NewDockerfileConnection(id uint32, conf *inventory.Config, asset *inventory
 	// this helps with running commands against the local connection
 	asset.Platform.Family = append(asset.Platform.Family, localFamily...)
 
+	if len(asset.GetConnections()) == 0 { // prevents panics by accesssing `asset.Connections[0]`
```

Comment:
asset.Connections should not be used here, use `conf` directly since it is passed in

Reply:
Should we be modifying `conf` instead of `asset` here? For connections, platformId, etc.?

Reply:
Yes, we should do:

```
conf.PlatformId = platformID
asset.PlatformIds = []string{platformID}
```

@imilchev Can you confirm that?

Reply:
I also think we should not do `if len(asset.GetConnections()) == 0 { // prevents panics by accesssing `asset.Connections[0]`. This case should never happen. If it happens, we need to fix the root cause instead of the symptom.

@imilchev please confirm

Reply:
connections can never be 0, this is confirmed. That's because the connection determines which provider to be called. If we ended up in a provider, we had a connection set

Reply:
for the question about

```go
conf.PlatformId = platformID
asset.PlatformIds = []string{platformID}
```

`conf` is actually equal (and also the same pointer as `asset.Connections[0]`. We can see that here https://github.com/mondoohq/cnquery/blob/main/providers/os/provider/provider.go#L305-L306

Reply:
Thank you for that context, I'll remove this if statement.